File: javascript/src/api_internal/autoconfig.ts
// this file is @generated

import {
  type AutoConfigSubscriptionOut,
  AutoConfigSubscriptionOutSerializer,
} from "../models/autoConfigSubscriptionOut";
import { AutoconfigDestination } from "./autoconfigDestination";
import { AutoconfigEndpoint } from "./autoconfigEndpoint";
import { HttpMethod, SvixRequest, type SvixRequestContext } from "../request";

export class Autoconfig {
  public constructor(private readonly requestCtx: SvixRequestContext) {}

  public get destination() {
    return new AutoconfigDestination(this.requestCtx);
  }

  public get endpoint() {
    return new AutoconfigEndpoint(this.requestCtx);
  }

  /** Get an AutoConfig subscription, including the bound endpoint or destination if any. */
  public async get(
    appId: string,
    autoconfigId: string
  ): Promise<AutoConfigSubscriptionOut> {
    const request = new SvixRequest(
      HttpMethod.GET,
      "/api/v1/app/{app_id}/autoconfig/{autoconfig_id}"
    );

    request.setPathParam("app_id", appId);
    request.setPathParam("autoconfig_id", autoconfigId);

    return await request.send(
      this.requestCtx,
      AutoConfigSubscriptionOutSerializer._fromJsonObject
    );
  }
}

File: rust/src/api_internal/autoconfig.rs
// this file is @generated
use super::{AutoconfigDestination, AutoconfigEndpoint};
use crate::{error::Result, models::*, Configuration};

pub struct Autoconfig<'a> {
    cfg: &'a Configuration,
}

impl<'a> Autoconfig<'a> {
    pub(super) fn new(cfg: &'a Configuration) -> Self {
        Self { cfg }
    }

    pub fn destination(&self) -> AutoconfigDestination<'a> {
        AutoconfigDestination::new(self.cfg)
    }

    pub fn endpoint(&self) -> AutoconfigEndpoint<'a> {
        AutoconfigEndpoint::new(self.cfg)
    }

    /// Get an AutoConfig subscription, including the bound endpoint or
    /// destination if any.
    pub async fn get(
        &self,
        app_id: String,
        autoconfig_id: String,
    ) -> Result<AutoConfigSubscriptionOut> {
        crate::request::Request::new(
            http::Method::GET,
            "/api/v1/app/{app_id}/autoconfig/{autoconfig_id}",
        )
        .with_path_param("app_id", app_id)
        .with_path_param("autoconfig_id", autoconfig_id)
        .execute(self.cfg)
        .await
    }
}
